File: app/src/main/java/com/tobiasfried/rasa/RecipeViewHolder.java
package com.tobiasfried.rasa;

import android.view.View;
import android.widget.TextView;

import com.tobiasfried.rasa.model.Recipe;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

public class RecipeViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.text_view_name)
    TextView nameTextView;

    @BindView(R.id.text_view_tea)
    TextView teaTextView;

    @BindView(R.id.text_view_sugar)
    TextView sugarTextView;

    @BindView(R.id.text_view_flavor)
    TextView flavorTextView;

    public RecipeViewHolder(@NonNull View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    public void bind(Recipe recipe) {
        // Set fields
        nameTextView.setText(recipe.getName());
        if (!recipe.getTeas().isEmpty()) {
            teaTextView.setText(recipe.getTeas().get(0).getName());
        }
        sugarTextView.setText(itemView.getContext().getResources().getStringArray(R.array.array_sugar_types)[recipe.getPrimarySweetener()]);
        if (!recipe.getIngredients().isEmpty()) {
            flavorTextView.setText(recipe.getIngredients().get(0).getName());
        }
    }

}

File: app/src/main/java/com/tobiasfried/rasa/recipes/RecipeViewHolder.kt
package com.tobiasfried.rasa.recipes

import android.view.View
import android.widget.TextView

import com.tobiasfried.rasa.domain.Recipe
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import com.tobiasfried.rasa.R

class RecipeViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

    @BindView(R.id.text_view_name)
    internal var nameTextView: TextView? = null

    @BindView(R.id.text_view_tea)
    internal var teaTextView: TextView? = null

    @BindView(R.id.text_view_sugar)
    internal var sugarTextView: TextView? = null

    @BindView(R.id.text_view_flavor)
    internal var flavorTextView: TextView? = null

    init {
        ButterKnife.bind(this, itemView)
    }

    fun bind(recipe: Recipe) {
        // Set fields
        nameTextView!!.text = recipe.name
        if (!recipe.teas.isEmpty()) {
            teaTextView!!.text = recipe.teas[0].name
        }
        sugarTextView!!.text = itemView.context.resources.getStringArray(R.array.array_sugar_types)[recipe.primarySweetener]
        if (!recipe.ingredients.isEmpty()) {
            flavorTextView!!.text = recipe.ingredients[0].name
        }
    }

}
